hudi-spark/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -64,6 +62,32 @@ public HoodieKey getKey(GenericRecord record) {
       throw new HoodieKeyException("Unable to find field names for record key or partition path in cfg");
     }
 
+    String recordKey = getRecordKey(record);
+    StringBuilder partitionPath = new StringBuilder();
+    for (String partitionPathField : partitionPathFields) {
+      partitionPath.append(getPartitionPath(record, partitionPathField));
+      partitionPath.append(DEFAULT_PARTITION_PATH_SEPARATOR);
+    }
+    partitionPath.deleteCharAt(partitionPath.length() - 1);
+
+    return new HoodieKey(recordKey, partitionPath.toString());
+  }
+
+  @Override
+  public String getPartitionPath(GenericRecord record, String partitionPathField) {
```

Comment:
I assume this whol;e change is just restructuring code

Reply:
Yes.